RHDEVDOCS-4741: Added information in various sections of 1.9 Pipeline… (#54712)
openshift/openshift-docs: modules/op-release-notes-1-9.adoc

[id="deprecated-features-1-9_{context}"]
== Deprecated and removed features

* In the Red Hat OpenShift Pipelines 1.9.0 release, `ClusterTasks` are deprecated and planned to be removed in a future release. As an alternative, you can use `Cluster Resolver`.

* In the Red Hat OpenShift Pipelines 1.9.0 release, the usage of the `triggers` and the `namespaceSelector` fields in a single `EventListener` specification is deprecated and planned to be removed in a future release. You can use these fields in different `EventListener` specifications successfully.

* In the Red Hat OpenShift Pipelines 1.9.0 release, the `tkn pipelinerun describe` command does not display timeouts for the `PipelineRun` resource.

* In the Red Hat OpenShift Pipelines 1.9.0 release, the the `PipelineResource` custom resource (CR) is deprecated. The `PipelineResource` CR was a Tech Preview feature and part of the `tekton.dev/v1alpha1` API.

> * In the Red Hat OpenShift Pipelines 1.9.0 release, custom image parameters from cluster tasks are deprecated. You must run an Operator-provided cluster task with it's own custom image and create a copy of the cluster task.

Reviewer: > You must run an Operator-provided cluster task with it's own custom image and create a copy of the cluster task.

You don't have to. If customers are OK with whatever we ship, they can use it out-of-the-box. In the case that they want to use their own image, they need to create a copy of cluster task and then use their own image in that copy.

Author: @ppitonak or @rupalibehera - Please confirm if my changes with respect to this comment are ok?

Reviewer: Nope, instead of 

>You must run an Operator-provided cluster task with it's own custom image and create a copy of the cluster task.

We should include something like

> As an alternative, you can copy a cluster task and use your custom image in it.

Author: Sure, thanks for providing feedback. I have made the change as you suggested. Please have a look.

Reviewer 2: @ppitonak please take a final look here

[id="breaking-changes-1-9_{context}"]
== Breaking changes

// .Pipelines

* With this update, the `Conditions` custom resource definition (CRD) type has been removed. As an alternative, use the WhenExpressions CRD instead.

* With this update, support for `tekton.dev/v1alpha1` API pipeline resources, such as, Pipeline, PipelineRun, Task, Clustertask, and TaskRun has been removed.

* With this update, the `tkn-pac setup` command has been removed. You can now re-use the functionality provided by the `tkn-pac setup` command by using the `tkn-pac webhook add` and `tkn-pac webhook update-token` commands. You can use the `tkn-pac webhook add` command to re-add a webhook to an existing Git repository. And, you can use the `tkn-pac webhook update-token` command to update the personal provider access token for an existing Secret object in the Git repository.

[id="deprecated-features-1-9_{context}"]
== Deprecated and removed features

* In the Red Hat OpenShift Pipelines 1.9 release, `ClusterTasks` are deprecated and planned to be removed in a future release. As an alternative, you can use `Cluster Resolver`.

> * In the Red Hat OpenShift Pipelines 1.9 release, the `Trigger` object and the `namespaceSelector` field in the `EventListener` specification are deprecated and planned to be removed in a future release.

Reviewer 3: `namespaceSelector` field in the `EventListener` specification are deprecated and planned to be removed in a future release.

From where did we get this?

Author: Hey @khrm, you had provided the following text in the Deprecated changes of the SME doc so I added this in the RN:

Deprecation Notices
Deprecation Warning: Having both Triggers as well as Namespace-Selector in EventListener Spec is deprecated.

Kindly suggest if you still want this text in the RN.

Reviewer 3: What I meant there is that you can't have Triggers as well as Namespace-Selector field at the same time.

Author: @khrm - Could you please provide me some more context with regards to this statement? The statement suggests that both these fields have been deprecated and I have stated the same. The new statement you provided is a little unclear to me.

Reviewer 3: What I meant is that you can't specify both fields in an EventListener at the same time. In future releases, it will fail the validation.
```
spec:
    namespaceSelector:...
    triggers:...
```
Spec above is deprecated.

Spec below are fine:
```
spec:
    namespaceSelector:
```

```
spec:
    triggers:
```

Reviewer 2: @khrm is trying to say that in a same eventlistener spec you cannot have both `namespaceSelector` and ` triggers` field.  The fields are not being deprecated but the usage of both the fields in a single spec file is deprecated, they can used in two different spec seperately.

Author: Thanks for the clarification @rupalibehera. I have made the change in the content. Kindly have a look and confirm if it looks good.

Reviewer 2: ```
In the Red Hat OpenShift Pipelines 1.9.0 release, the usage of the `triggers` and  `namespaceSelector` field in a single `EventListener` specification is deprecated and planned to be removed in a future release. You can use these fields in different `EventListener` specifications successfully
```

Reviewer 3: What about ``the simulataneous usage of  the `triggers` and  `namespaceSelector` field in a single `EventListener` specification is deprecated and planned to be removed in a future release. You can use these fields in different `EventListener` specifications ``

> * In the Red Hat OpenShift Pipelines 1.9 release, the `Trigger` object and the `namespaceSelector` field in the `EventListener` specification are deprecated and planned to be removed in a future release.

Reviewer 4: ```suggestion
* In the Red Hat OpenShift Pipelines 1.9.0 release, the `Trigger` object and the `namespaceSelector` field in the `EventListener` specification are deprecated and planned to be removed in a future release.
```

Author: Thanks for the suggestion. Fixed.

// Vincent Demeester

* Before this update, if duplicate secrets are present in a service account for a pipelines run, it resulted in failure in task pod creation. With this update, this issue is resolved and the task pod is created successfully even if duplicate secrets were present in a service account.
// Vincent Demeester

* Before this update, by looking at the TaskRun's `spec.StatusMessage` field, users could not distinguish whether the `TaskRun` had been cancelled by the user or by a `PipelineRun` that was part of it. With this update, this issue is resolved and users can distinguish the status of the `TaskRun` by looking at the TaskRun's `spec.StatusMessage` field.
// Vincent Demeester

* With this update, the user can specify tasks or timeout.
// Vincent Demeester

* Before this update, a user could not remove webhook validation when deleting old versions of invalid objects. With this update, this issue is resolved and user can remove webhook validation when deleting old versions of invalid objects.
// Vincent Demeester @vdeemester

* With this update, the user can merge labels and annotations instead of override.
// Vincent Demeester @vdeemester

// .Triggers

> * Before this update, log keys did not have same keys as in pipelines controllers. With this update, this issue has been resolved and the log keys have been updated to match the log stream. The keys in logs have been changed from "ts" to "timestamp", from "level" to "severity", and "message" to "msg".

Reviewer 3: ``log keys have been updated to match the log stream`` 
Should it be to match the log stream of pipelines controllers?

Author: If you say it should be pipelines controllers, I can make a change. The text provided in the SME doc is as follows:

Update log keys to match logstream:  Log lines formatted as JSON have the severity in "severity" (was "level"), timestamp in "timestamp" (was "ts"), and message in "message" (was "msg").

Nowhere is it mentioned to match the log stream of pipeline controllers. Please suggest.

Reviewer 3: Yes, it was to match the log stream of pipeline controllers. Triggers and Pipelines have different keys, this will make it same for both.

Author: Thanks for the clarification, @khrm. I have made the change as per your suggestion. Kindly confirm.

// https://issues.redhat.com/browse/SRVKP-1959
// @KhurramBaig

// .CLI

> * Before this update, if a pull request was deleted with an unknown status, an error message was not generated. With this update, this issue is resolved and an error message is generated.

Reviewer 5: PR does not mean pull request, its PipelineRun

Author: Fixed as per your suggestion. Thanks.

// https://github.com/tektoncd/cli/pull/1684
// Piyush Garg @piyush-garg

> * Before this update, the `kubeconfig` file with bundle commands like `list`, `push` was required to access the cluster. With this update, this issue has been resolved and the `kubeconfig` file is not required to access the cluster.

Reviewer 5: do not use - `to access the cluster`

Author: Thanks for the suggestion. Fixed.

// https://github.com/tektoncd/cli/pull/1557
// Piyush Garg @piyush-garg

> * Before this update, if the parent pipelinerun is running while deleting TaskRuns, then TaskRuns will be deleted. With this update, this issue issue is resolved and TaskRuns are not getting deleted if the parent PipelineRun is running.

Reviewer 5: issue is written twice

Author: The issue is not repeated twice as you suggested, @piyush-garg. I have simply stated what happens before and after this update. Could you confirm this once again?

Reviewer 2: ```suggestion
* Before this update, if the parent pipelinerun is running while deleting TaskRuns, then TaskRuns will be deleted. With this update, this issue  is resolved and TaskRuns are not getting deleted if the parent PipelineRun is running.
```

Reviewer 2: @Dhruv-Soni11 please check the above comment

// https://github.com/tektoncd/cli/pull/1736
// Piyush Garg @piyush-garg

> * With this command, a checkbox has been for the set bundle max field used in the pipeline controller.

Reviewer 5: i dont know what is this

Author: Hi @piyush-garg , this text is added with respect to the following: 

Adding a check for the set bundle max used in the pipeline controller

May I know if this text is a part of this overall issue:

Adding a check for the set bundle max used in the pipeline controller
Tkn cli will error if attempting to create a bundle with more objects than allowed by the pipeline controller.

If that is the case, I have added a different entry for the same. It is as follows:

 Before this update, if the user attempted to create a bundle with more objects than the pipeline controller permitted, the Tekton CLI did not display an error message. With this update, this issue is resolved and the Tekton CLI displays an error message message if the user added more objects than the limit permitted in the pipeline.
 
 If you suggest, I could remove this one line entry I added here and only retain the paragraph added above.

Reviewer 5: please remove this

Reviewer 2: @Dhruv-Soni11 please remove this single line entry `With this command, a checkbox has been for the set bundle max field used in the pipeline controller.` and just keep this `Before this update, if the user attempted to create a bundle with more objects than the pipeline controller permitted, the Tekton CLI did not display an error message. With this update, this issue is resolved and the Tekton CLI displays an error message if the user added more objects than the limit permitted in the pipeline.`

Author: Got it @rupalibehera. I will remove this line and keep the content as you suggested.

// Piyush Garg @piyush-garg

> * Before this update, if the user attempted to create a bundle with more objects than the pipeline controller permitted, the Tekton CLI did not display an error message. With this update, this issue is resolved and the Tekton CLI displays an error message message if the user added more objects than the limit permitted in the pipeline.

Reviewer 5: message is written twice

Author: Again, the text is added in different context in terms of a before or after perspective. Could you read this again to confirm?

Reviewer 2: @Dhruv-Soni11 , Piyush means that the word message is repeated twice

Reviewer 2: > @Dhruv-Soni11 , Piyush means that the word `message` is repeated twice in third line `Tekton CLI displays an error message message`

Author: Got it, thanks for pointing that out @piyush-garg and @rupalibehera. I have made this change in the content.

> * Before this update, if the user attempted to create a bundle with more objects than the pipeline controller permitted, the Tekton CLI did not display an error message. With this update, this issue is resolved and the Tekton CLI displays an error message message if the user added more objects than the limit permitted in the pipeline.

Reviewer 5: tekton cli desplays error message during building the bundle

Author: Thank you for the suggestion. Fixed.

> * Before this update, if the user attempted to create a bundle with more objects than the pipeline controller permitted, the Tekton CLI did not display an error message. With this update, this issue is resolved and the Tekton CLI displays an error message message if the user added more objects than the limit permitted in the pipeline.

Reviewer 5: limit permitted in pipeline controller

// Module included in the following assembly:
//
// * cicd/pipelines/op-release-notes.adoc
:_content-type: REFERENCE
[id="op-release-notes-1-9_{context}"]
= Release notes for {pipelines-title} General Availability 1.9

With this update, {pipelines-title} General Availability (GA) 1.9 is available on {product-title} 4.11 and later versions.

[id="breaking-changes-1-9_{context}"]
== Breaking changes

// .Pipelines

> * With this update, the `Conditions` custom resource definition (CRD) type has been removed. As an alternative, use the WhenExpressions instead.

Author: The SME suggested not to use the term CRD in the and so I will lean towards avoiding it. But I will definitely make the other change you suggested. Thanks.

> * With this update, the `Conditions` custom resource definition (CRD) type has been removed. As an alternative, use the WhenExpressions instead.

Reviewer 6: OK, I see it is not a CRD, just a key "when:" in tasks

> * With this update, support for `tekton.dev/v1alpha1` API pipeline resources, such as, Pipeline, PipelineRun, Task, Clustertask, and TaskRun has been removed.

Author: Fixed.

> * With this update, the `tkn-pac setup` command has been removed. You can now re-use the functionality provided by the `tkn-pac setup` command by using the `tkn-pac webhook add` and `tkn-pac webhook update-token` commands. You can use the `tkn-pac webhook add` command to re-add a webhook to an existing Git repository. And, you can use the `tkn-pac webhook update-token` command to update the personal provider access token for an existing Secret object in the Git repository.

Author: Fixed.

[id="deprecated-features-1-9_{context}"]
== Deprecated and removed features

* In the Red Hat OpenShift Pipelines 1.9.0 release, `ClusterTasks` are deprecated and planned to be removed in a future release. As an alternative, you can use `Cluster Resolver`.

> * In the Red Hat OpenShift Pipelines 1.9.0 release, the usage of the `triggers` and the `namespaceSelector` fields in a single `EventListener` specification is deprecated and planned to be removed in a future release. You can use these fields in different `EventListener` specifications successfully.

Author: Fixed.

> * In the Red Hat OpenShift Pipelines 1.9.0 release, the usage of the `triggers` and the `namespaceSelector` fields in a single `EventListener` specification is deprecated and planned to be removed in a future release. You can use these fields in different `EventListener` specifications successfully.

Reviewer 6: I don't see any change here - and maybe replace "usage" with "use" to simplify

* In the Red Hat OpenShift Pipelines 1.9.0 release, the `tkn pipelinerun describe` command does not display timeouts for the `PipelineRun` resource.

* In the Red Hat OpenShift Pipelines 1.9.0 release, the the `PipelineResource` custom resource (CR) is deprecated. The `PipelineResource` CR was a Tech Preview feature and part of the `tekton.dev/v1alpha1` API.

> * In the Red Hat OpenShift Pipelines 1.9.0 release, custom image parameters in cluster tasks are deprecated. You must run an operator-provided cluster task with a custom image. The following parameters must be configured in the custom image:

Author: Fixed.

+
[source,yaml]
----
buildah task doesn't expose "BUILDER_IMAGE" param
git-clone task doesn't expose "gitInitImage" param
kn task doesn't expose "kn-image" param
kn-apply task doesn't expose "KN_IMAGE" param
maven task doesn't expose "MAVEN_IMAGE" param
s2i-* tasks don't expose "BUILDER_IMAGE" param
tkn task doesn't expose "TKN_IMAGE" param

> ----

Reviewer: This section is confusing - those 7 parameters are still in cluster tasks shipped with 1.9 but they will be removed in 1.10.

> You must run an operator-provided cluster task with a custom image
This doesn't make sense. Actually, if a customer wants to use a custom image, they nee d to clone a cluster task and update the image in that newly cloned task.

Reviewer 2: custom image param (for eg: in tnk task `TKN_IMAGE` is a custom image param) from cluster tasks are deprecated.

Author: @rupalibehera and @ppitonak - I've made some suggestions as per your comments. Kindly have a look and let me know what you think.

[id="known-issues-1-9_{context}"]
== Known issues

// .Operator

> * The `chains-secret` and `chains-config` config map are removed after you uninstall the Red Hat OpenShift Pipelines operator. As they contain user data, they should be preserved and not deleted.

Author: Fixed.

// https://issues.redhat.com/browse/SRVKP-2396

// .PAC

> * When running the `tkn pac` set of commands on Windows, you may receive the following error message: Command finished with error: not supported by Windows.

Author: Fixed.

> 

Author: Fixed.

> 

Author: Fixed.

Workaround: Set the `NO_COLOR` environment variable to `true`.
// https://issues.redhat.com/browse/SRVKP-2197

* Running the `tkn pac resolve -f <filename> | oc create -f` command may not provide expected results, if the `tkn pac resolve` command uses a templated parameter value to function.

> 

Author: Fixed.

Workaround: To mitigate this issue, save the output of `tkn pac resolve` in a temporary file by running the `tkn pac resolve -f <filename> > tempfile.yaml` command and then run the `oc create -f tempfile.yaml` command. For example, `tkn pac resolve -f <filename> -o /tmp/pull-request-resolved.yaml && kubectl create -f /tmp/pull-request-resolved.yaml`.

[id="fixed-issues-1-9_{context}"]
== Fixed issues

// .Pipelines

> * Before this update, after replacing an empty array, the original array returns an empty string rendering the paramaters inside the invalid. With this update, this issue is resolved and the original array return as empty. 

Reviewer 6: " rendering the paramaters inside the invalid. "

This doesn't make sense - maybe remove "the" before invalid, but not sure if that is technically correct

Author: Fixed it as you suggested, I think it is technically correct. SME has already looked at this.

> * Before this update, after replacing an empty array, the original array returns an empty string rendering the paramaters inside the invalid. With this update, this issue is resolved and the original array return as empty. 

Reviewer 6: ```suggestion
* Before this update, after replacing an empty array, the original array returned an empty string rendering the paramaters inside the invalid. With this update, this issue is resolved and the original array returns as empty. 
```

Author: Fixed.

// Vincent Demeester

> * Before this update, if duplicate secrets are present in a service account for a pipelines run, it resulted in failure in task pod creation. With this update, this issue is resolved and the task pod is created successfully even if duplicate secrets were present in a service account.

Reviewer 6: ```suggestion
* Before this update, if duplicate secrets were present in a service account for a pipelines run, it resulted in failure in task pod creation. With this update, this issue is resolved and the task pod is created successfully even if duplicate secrets are present in a service account.
```

Author: Fixed.

// Vincent Demeester

* Before this update, by looking at the TaskRun's `spec.StatusMessage` field, users could not distinguish whether the `TaskRun` had been cancelled by the user or by a `PipelineRun` that was part of it. With this update, this issue is resolved and users can distinguish the status of the `TaskRun` by looking at the TaskRun's `spec.StatusMessage` field.
// Vincent Demeester

> * Before this update, you could not set some fields of `timeouts`. With this update this issue is resolved, and you can specify some fields of `timeouts` and let the rest of the fields be default.

Reviewer 6: ```suggestion
* Before this update, you could not set some fields of `timeouts`. With this update, this issue is resolved and you can specify some fields of `timeouts` and let the rest of the fields be default.
```

Author: Fixed.

> * Before this update, you could not set some fields of `timeouts`. With this update this issue is resolved, and you can specify some fields of `timeouts` and let the rest of the fields be default.

Reviewer 6: This sentence is very hard to understand. Is it certain fields that couldn't be set, or was it an "all-or-nothing" case? this needs clarification

Author: As best as I can understand, it means that only some fields can be set and the rest are default.

Author: @rupalibehera - Could you please take a look at this comment if it is technically accurate?

Reviewer 2: @vdemeester can you please take a look here

// Vincent Demeester

* Before this update, webhook validation was removed on deletion of old versions of invalid objects. With this update, this issue is resolved.
// Vincent Demeester @vdeemester

> * Before this update, a race condition may occur if another tool updates labels or annotations on a PipelineRun or TaskRun. With this update, this issue is resolved and you can merge labels or annotations.

Reviewer 6: ```suggestion
* Before this update, a race condition could occur if another tool updated labels or annotations on a PipelineRun or TaskRun. With this update, this issue is resolved and you can merge labels or annotations.
```

Author: Fixed.

// Vincent Demeester @vdeemester

// .Triggers

> * Before this update, log keys did not have same keys as in pipelines controllers. With this update, this issue has been resolved and the log keys have been updated to match the log stream of pipeline controllers. The keys in logs have been changed from "ts" to "timestamp", from "level" to "severity", and "message" to "msg".

Reviewer 6: ```suggestion
* Before this update, log keys did not have the same keys as in pipelines controllers. With this update, this issue has been resolved and the log keys have been updated to match the log stream of pipeline controllers. The keys in logs have been changed from "ts" to "timestamp", from "level" to "severity", and from "message" to "msg".
```

Author: Fixed.

// https://issues.redhat.com/browse/SRVKP-1959
// @KhurramBaig

// .CLI

* Before this update, if a PipelineRun was deleted with an unknown status, an error message was not generated. With this update, this issue is resolved and an error message is generated.
// https://github.com/tektoncd/cli/pull/1684
// Piyush Garg @piyush-garg

* Before this update, to access bundle commands like `list` and `push`, it was required to use the `kubeconfig` file . With this update, this issue has been resolved and the `kubeconfig` file is not required to access bundle commands.
// https://github.com/tektoncd/cli/pull/1557
// Piyush Garg @piyush-garg

> * Before this update, if the parent pipelinerun is running while deleting TaskRuns, then TaskRuns will be deleted. With this update, this issue is resolved and TaskRuns are not getting deleted if the parent PipelineRun is running.

Reviewer 6: ```suggestion
* Before this update, if the parent pipelinerun was running while deleting TaskRuns, then TaskRuns would be deleted. With this update, this issue is resolved and TaskRuns are not getting deleted if the parent PipelineRun is running.
```

// https://github.com/tektoncd/cli/pull/1736
// Piyush Garg @piyush-garg

> * Before this update, if the user attempted to build a bundle with more objects than the pipeline controller permitted, the Tekton CLI did not display an error message. With this update, this issue is resolved and the Tekton CLI displays an error message if the user attempted to build a bundle with more objects than the limit permitted in the pipeline controller.

Reviewer 6: ```suggestion
* Before this update, if the user attempted to build a bundle with more objects than the pipeline controller permitted, the Tekton CLI did not display an error message. With this update, this issue is resolved and the Tekton CLI displays an error message if the user attempts to build a bundle with more objects than the limit permitted in the pipeline controller.
```

// https://github.com/tektoncd/cli/pull/1572
// Piyush Garg @piyush-garg

// .Operator

* Before this update, if namespaces were removed from the cluster, then the operator did not remove namespaces from the `ClusterInterceptor ClusterRoleBinding` subjects. With this update, this issue has been resolved, and the operator removes the namespaces from the `ClusterInterceptor ClusterRoleBinding` subjects.
// Shubham Minglani

> * Before this update, the default installation of the Red Hat OpenShift Pipelines Operator resulted in the `pipelines-scc-rolebinding security context constraint` (SCC) role binding resource remaining in the cluster. With this update, the default installation of the Red Hat OpenShift Pipelines Operator results in the `pipelines-scc-rolebinding security context constraint` (SCC) role binding resource resource to be removed from the cluster.

Author: Fixed.

// https://github.com/tektoncd/operator/pull/1156
// https://issues.redhat.com/browse/SRVKP-2520
// Shubham Minglani

// .PAC

> * Before this update, {pac} did not get updated values from the {pac} configmap. With this update, this issue is fixed and {pac} configmap looks for any new changes.

Reviewer 6: Probably should be ticked `ConfigMap` as per doc contrib guikdelines

// Savita Ashture

> * Before this update, {pac} controller did not wait for the `tekton.dev/pipeline` label to be updated and added the `checkrun id` label, which would cause race conditions. With this update, a fix has been added to the {pac} controller to wait for the `tekton.dev/pipeline` label to be updated and then add the `checkrun id` label, which helps to avoid race conditions.

Author: Fixed.

// Savita Ashture

> * Before this update, the `tkn-pac create repo` command did not override a `PipelineRun` if it already existed in the Git Repository. With this update, `tkn-pac create` command is fixed to override a `PipelineRun` if it exists in the Git Repository and this resolves the issue successfully.

Reviewer 6: I am not sure if Git Repository should be upper case - in general, it should not, but maybe the conventions here are different

Author: Fixed.

// Savita Ashture

> * Before this update, the `tkn pac describe` command did not display reasons for every message. With this update, this issue is fixed and the `tkn pac describe` command  and displays reasons for every message.

Author: Fixed.

// Savita Ashture

> * Before this update, a pull request failed if the user in the annotation provided values by using a regex form, for example, `refs/head/rel-*`. The pull request failed because it was missing `refs/heads` in its base branch. With this update, we add the prefix and check if it matches. This resolves the issue and the pull request is successful.

Author: Fixed.

// Module included in the following assembly:
//
// * cicd/pipelines/op-release-notes.adoc
:_content-type: REFERENCE
[id="op-release-notes-1-9_{context}"]
= Release notes for {pipelines-title} General Availability 1.9

With this update, {pipelines-title} General Availability (GA) 1.9 is available on {product-title} 4.10, and is planned to be available on {product-title} 4.11 and later.

[id="breaking-changes-1-9_{context}"]
== Breaking changes

// .Pipelines

> * With this update, the `Conditions` custom resource definition (CRD) type has been removed. As an alternative, use the WhenExpressions CRD instead.

Author: Thanks for the suggestion. Fixed.

* In the Red Hat OpenShift Pipelines 1.9 release, the the `PipelineResource` custom resource (CR) is deprecated. The `PipelineResource` CR was a Tech Preview feature and part of the `tekton.dev/v1alpha1` API.

* In the Red Hat OpenShift Pipelines 1.9 release, custom image parameters in cluster tasks are deprecated. You must run an operator-provided cluster task with a custom image. The following parameters must be configured in the custom image:

+
[source,yaml]
----
buildah task doesn't expose "BUILDER_IMAGE" param
git-clone task doesn't expose "gitInitImage" param
kn task doesn't expose "kn-image" param
kn-apply task doesn't expose "KN_IMAGE" param
maven task doesn't expose "MAVEN_IMAGE" param
s2i-* tasks don't expose "BUILDER_IMAGE" param
tkn task doesn't expose "TKN_IMAGE" param
----

[id="known-issues-1-9_{context}"]
== Known issues

// .Operator

* The `chains-secret` and `chains-config` config map are removed after you uninstall the Red Hat OpenShift Pipelines operator. As they contain user data, they should be preserved and not deleted.
// https://issues.redhat.com/browse/SRVKP-2396

// .PAC

> * When running the `tkn pac` set of commands on Windows, you may receive the following error message: Command finished with error: not supported by Windows.

Author: Hi @concaf - I have also updated the content with these set of known issues that you suggested some time ago, could you please take a look at these as well?

Reviewer 4: looks good, thanks :)

Author: Thanks.

Workaround: Set the `NO_COLOR` environment variable to `true`.
// https://issues.redhat.com/browse/SRVKP-2197

* Running the `tkn pac resolve -f <filename> | oc create -f` command may not provide expected results, if the `tkn pac resolve` command uses a templated parameter value to function.

> Workaround: To mitigate this issue, save the output of `tkn pac resolve` in a temporary file by running the `tkn pac resolve -f <filename> > tempfile.yaml` command and then run the `oc create -f tempfile.yaml` command.

Reviewer 7: how about providing a sample example like this 

```
tkn pac resolve -f <filename> -o /tmp/pull-request-resolved.yaml && kubectl create -f /tmp/pull-request-resolved.yaml
```

[id="fixed-issues-1-9_{context}"]
== Fixed issues

// .Pipelines

* Before this update, after replacing an empty array, the original array returns an empty string rendering the paramaters inside the invalid. With this update, this issue is resolved and the original array return as empty. 
// Vincent Demeester

* Before this update, if duplicate secrets are present in a service account for a pipelines run, it resulted in failure in task pod creation. With this update, this issue is resolved and the task pod is created successfully even if duplicate secrets were present in a service account.
// Vincent Demeester

* Before this update, by looking at the TaskRun's `spec.StatusMessage` field, users could not distinguish whether the `TaskRun` had been cancelled by the user or by a `PipelineRun` that was part of it. With this update, this issue is resolved and users can distinguish the status of the `TaskRun` by looking at the TaskRun's `spec.StatusMessage` field.
// Vincent Demeester

> * Before this update, the user could not validate timeouts. With this update, this issue is resolved and the user can specify timeouts.

Reviewer 8: This is more about user could not set "only" some fields of the `timeouts` and let the rest be default. Now it's possible.

Author: Thanks for the suggestion. Fixed.

// Vincent Demeester

> * Before this update, the user could not remove webhook validation when deleting old versions of invalid objects. With this update, this issue is resolved and user can remove webhook validation.

Reviewer 8: The user doesn't do anything here. Prior to this update, validation was run on deletion (which caused problems with old objects), now it doesn't run anymore.

Author: Thanks for the suggestion. Fixed.

// Vincent Demeester @vdeemester

> * With this update, the user can merge labels and annotations instead of override.

Author: Thanks for the suggestion. Fixed.

* In the Red Hat OpenShift Pipelines 1.9.0 release, the the `PipelineResource` custom resource (CR) is deprecated. The `PipelineResource` CR was a Tech Preview feature and part of the `tekton.dev/v1alpha1` API.

* In the Red Hat OpenShift Pipelines 1.9.0 release, custom image parameters from cluster tasks are deprecated. As an alternative, you can copy a cluster task and use your custom image in it.

[id="known-issues-1-9_{context}"]
== Known issues

// .Operator

* The `chains-secret` and `chains-config` config maps are removed after you uninstall the Red Hat OpenShift Pipelines Operator. As they contain user data, they should be preserved and not deleted.
// https://issues.redhat.com/browse/SRVKP-2396

// .PAC

* When running the `tkn pac` set of commands on Windows, you may receive the following error message: `Command finished with error: not supported by Windows.`
+

Workaround: Set the `NO_COLOR` environment variable to `true`.
// https://issues.redhat.com/browse/SRVKP-2197

* Running the `tkn pac resolve -f <filename> | oc create -f` command may not provide expected results, if the `tkn pac resolve` command uses a templated parameter value to function.
+

> Workaround: To mitigate this issue, save the output of `tkn pac resolve` in a temporary file by running the `tkn pac resolve -f <filename> > tempfile.yaml` command and then run the `oc create -f tempfile.yaml` command. For example, `tkn pac resolve -f <filename> -o /tmp/pull-request-resolved.yaml && kubectl create -f /tmp/pull-request-resolved.yaml`.

Reviewer 9: `tkn pac resolve -f <filename> > tempfile.yaml` does not work, we need to change it to `tkn pac resolve -f <filename> -o tempfile.yaml`

Author: @VeereshAradhya - Hi, Thanks for looking at my PR. I hope you have looked at it thoroughly and this is the only comment you have. I hope there aren't any other comments which I need to incorporate. Please confirm.

Reviewer 9: Yes, only this comment needs to be addressed

// Vincent Demeester @vdeemester

// .Triggers

* Before this update, log keys did not have the same keys as in pipelines controllers. With this update, this issue has been resolved and the log keys have been updated to match the log stream of pipeline controllers. The keys in logs have been changed from "ts" to "timestamp", from "level" to "severity", and from "message" to "msg".
// https://issues.redhat.com/browse/SRVKP-1959
// @KhurramBaig

// .CLI

* Before this update, if a PipelineRun was deleted with an unknown status, an error message was not generated. With this update, this issue is resolved and an error message is generated.
// https://github.com/tektoncd/cli/pull/1684
// Piyush Garg @piyush-garg

* Before this update, to access bundle commands like `list` and `push`, it was required to use the `kubeconfig` file . With this update, this issue has been resolved and the `kubeconfig` file is not required to access bundle commands.
// https://github.com/tektoncd/cli/pull/1557
// Piyush Garg @piyush-garg

> * Before this update, if the parent pipelinerun was running while deleting TaskRuns, then TaskRuns would be deleted. With this update, this issue is resolved and TaskRuns are not getting deleted if the parent PipelineRun is running.

Reviewer 9: @piyush-garg we need to inform users about the `--ignore-running-pipelinerun` right?

Reviewer 5: Yes, that is added in features

Reviewer 9: ok. :+1:
